tests/migration_test.go

```diff
@@ -2308,6 +2309,68 @@ var _ = Describe("[Serial][rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][leve
 		})
 
 	})
+
+	Context("With Huge Pages", func() {
+		var hugepagesVmi *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			hugepagesVmi = tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
+		})
+
+		table.DescribeTable("should consume hugepages ", func(hugepageSize string, memory string) {
+			hugepageType := k8sv1.ResourceName(k8sv1.ResourceHugePagesPrefix + hugepageSize)
+			v, err := cluster.GetKubernetesVersion()
+			Expect(err).ShouldNot(HaveOccurred())
+			if strings.Contains(v, "1.16") {
+				hugepagesVmi.Annotations = map[string]string{
+					v1.MemfdMemoryBackend: "false",
+				}
+				log.DefaultLogger().Object(hugepagesVmi).Infof("Fall back to use hugepages source file. Libvirt in the 1.16 provider version doesn't support memfd as memory backend")
+			}
+
+			count := 0
+			nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+			for _, node := range nodes.Items {
+				if v, ok := node.Status.Capacity[hugepageType]; ok && !v.IsZero() {
+					count += 1
+				}
+			}
+
+			if count < 2 {
+				Skip(fmt.Sprintf("Not enough nodes with hugepages %s capacity. Need 2, found %d.", hugepageType, count))
+			}
+
+			hugepagesVmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse(memory)
+
+			hugepagesVmi.Spec.Domain.Memory = &v1.Memory{
+				Hugepages: &v1.Hugepages{PageSize: hugepageSize},
+			}
+
+			By("Starting hugepages VMI")
+			_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(hugepagesVmi)
+			Expect(err).ToNot(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(hugepagesVmi)
+
+			By("starting the migration")
+			migration := tests.NewRandomMigration(hugepagesVmi.Name, hugepagesVmi.Namespace)
+			migrationUID := tests.RunMigrationAndExpectCompletion(virtClient, migration, tests.MigrationWaitTime)
+
+			// check VMI, confirm migration state
+			tests.ConfirmVMIPostMigration(virtClient, hugepagesVmi, migrationUID)
+
+			// delete VMI
+			By("Deleting the VMI")
+			Expect(virtClient.VirtualMachineInstance(hugepagesVmi.Namespace).Delete(hugepagesVmi.Name, &metav1.DeleteOptions{})).To(Succeed())
+
+			By("Waiting for VMI to disappear")
+			tests.WaitForVirtualMachineToDisappearWithTimeout(hugepagesVmi, 240)
+		},
+			table.Entry("[Serial][test_id:1671]hugepages-2Mi", "2Mi", "2Mi"),
+			table.Entry("[Serial][test_id:1672]hugepages-1Gi", "1Gi", "1Gi"),
```

Comment:
i don't think our ci has nodes with 1Gi huge pages enabled by default. so realistically this might never execute.  maybe that's not an issue and there are other environments where 1Gi huge pages exist and will exercise this. I don't think our standard CI environment will though.

Reply:
+1 by default we are configuring 64 pages of 2MB pages size.

Reply:
Understood and agreed. Though our CI isn't configured to explicitly test this, other environments might be able to make use of it.